src/NLog/LayoutRenderers/IdentityLayoutRenderer.cs

```diff
@@ -95,36 +95,45 @@ public IdentityLayoutRenderer()
         /// <param name="logEvent">Logging event.</param>
         protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         {
-            IPrincipal principal = System.Threading.Thread.CurrentPrincipal;
-            if (principal != null)
+            IIdentity identity = GetValue();
+            if (identity != null)
             {
-                IIdentity identity = principal.Identity;
-                if (identity != null)
+                string separator = string.Empty;
+
+                if (IsAuthenticated)
+                {
+                    builder.Append(separator);
+                    separator = Separator;
+
+                    builder.Append(identity.IsAuthenticated ? "auth" : "notauth");
+                }
+
+                if (AuthType)
+                {
+                    builder.Append(separator);
+                    separator = Separator;
+                    builder.Append(identity.AuthenticationType);
+                }
+
+                if (Name)
                 {
-                    string separator = string.Empty;
-
-                    if (IsAuthenticated)
-                    {
-                        builder.Append(separator);
-                        separator = Separator;
-
-                        builder.Append(identity.IsAuthenticated ? "auth" : "notauth");
-                    }
-
-                    if (AuthType)
-                    {
-                        builder.Append(separator);
-                        separator = Separator;
-                        builder.Append(identity.AuthenticationType);
-                    }
-
-                    if (Name)
-                    {
-                        builder.Append(separator);
-                        builder.Append(identity.Name);
-                    }
+                    builder.Append(separator);
+                    builder.Append(identity.Name);
                 }
             }
+
+        }
+
+        /// <inheritdoc />
+        object IRawValue.GetRawValue(LogEventInfo logEvent)
```

Comment:
@snakefoot do you think this layout renderer should implement IRawValue, and if so, should use the options IsAuthenticated/AuthType/Name is the rawValue? (well then it's a string)

Reply:
OK thanks! done